azext_iot/central/providers/device_provider.py

```diff
@@ -240,3 +240,10 @@ def _dps_populate_essential_info(self, dps_info, device_status: DeviceStatus):
             "error": error.get(device_status),
         }
         return filtered_dps_info
+
+    def generate_device_key(
+        self, masterkey, device_id,
+    ):
+        return central_services.auth.generate_device_key(
```

Comment:
this doesnt need to be in the service layer since it isnt making any network calls